src/main/javascript/src/pages/application/view/view.js

```diff
@@ -77,6 +63,37 @@ export class ApplicationView {
 		});
 	}
 
+	getApplicationsByFilter(){
+		let filter = this.statusFilter
+
+		this.applicationManager.getApplicationsForProf(this.currentUser.id).then(response => {
+			this.requested = [];
+			response.forEach((application, index) => {
+				application.showDetails = false;
+				application.detailToggleId = application.id + "requested"
+				this.userManager.getUserByApplication(this.currentUser.id).then(response => {
+					application.applicant = response[0];
+					if(filter === "all" || application.status === filter){
```

Comment:
I'm a little sad the .filter() function didn't get used, but this works well.

No changes required